### PR TITLE
New API for authorized-apps

### PR DESCRIPTION
**Pull request by somindatommy**
Part of the fix for https://github.com/wso2/product-is/issues/8387

## Approach
Introducing `v2` apis for `authorized-apps`. Compared to the previous version, the noticeable change is in the response. No change has been done for the core of the implementation. ApplicationManagementService has been used to retrieve resourceId of the application.

**v1**
```
"appId": "Self-Care-Test",
"clientId": "dicug2QkfSPHTV_zjJgc5lNGeHMa"
```

**v2**
```
"id": "67fc2c40-a351-4bee-87e3-dd1d26fd6e51",
"name": "pickup-dispatch",
"clientId": "AU_MPjPv6PC7czrnll6E5S3pyiQa"
```

## Follow-up actions
- [x] Merge PR https://github.com/wso2/carbon-identity-framework/pull/3224
- [ ] Merge PR https://github.com/wso2/identity-rest-dispatcher/pull/295

### CODE REVIEW

File: components/org.wso2.carbon.identity.api.user.authorized.apps/org.wso2.carbon.identity.rest.api.user.authorized.apps.v2/pom.xml [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>${maven.compiler.plugin.version}</version>
+                <configuration>
+                    <source>1.8</source>
+                    <target>1.8</target>
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>org.wso2.maven.plugins</groupId>
```
Comment (IsuraD) [inline components/org.wso2.carbon.identity.api.user.authorized.apps/org.wso2.carbon.identity.rest.api.user.authorized.apps.v2/pom.xml:43]: comment this

File: components/org.wso2.carbon.identity.api.user.authorized.apps/org.wso2.carbon.identity.rest.api.user.authorized.apps.v2/pom.xml [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.testng</groupId>
+            <artifactId>testng</artifactId>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
+
+</project>
```
Comment (IsuraD) [inline components/org.wso2.carbon.identity.api.user.authorized.apps/org.wso2.carbon.identity.rest.api.user.authorized.apps.v2/pom.xml:173]: add a new line end of a file

Comment (somindatommy) [reply components/org.wso2.carbon.identity.api.user.authorized.apps/org.wso2.carbon.identity.rest.api.user.authorized.apps.v2/pom.xml]: Fixed in f393d65e6020758087f6ab52c75b5c5533a7716b

File: components/org.wso2.carbon.identity.api.user.authorized.apps/org.wso2.carbon.identity.rest.api.user.authorized.apps.v2/pom.xml (hunk 1 above)
Comment (somindatommy) [reply components/org.wso2.carbon.identity.api.user.authorized.apps/org.wso2.carbon.identity.rest.api.user.authorized.apps.v2/pom.xml:43]: Fixed in f393d65e6020758087f6ab52c75b5c5533a7716b

**Review by IsuraD — APPROVED** [review]

**Pull request merged**